extensions/oidc-client-registration/deployment/src/main/java/io/quarkus/oidc/client/registration/deployment/OidcClientRegistrationBuildTimeConfig.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.oidc.client.registration.deployment;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * Build time configuration for OIDC client registration.
+ */
+@ConfigRoot
+public class OidcClientRegistrationBuildTimeConfig {
+    /**
+     * If the OIDC client registration extension is enabled.
+     */
+    @ConfigItem(defaultValue = "true")
+    public boolean enabled;
+}
```

Comment:
Use the interface pattern instead (see https://quarkus.io/guides/writing-extensions#configuration)

Reply:
Sure, good point, it will be fixed as part of #39185, but I'm not certain if #39185 must be done first